std/string.d

```diff
@@ -2840,15 +2841,17 @@ if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range) &&
         auto c = input.front;
         if (std.ascii.isASCII(c))
         {
-            if (!std.ascii.isWhite(c))
+            if ((chars == "" && !std.ascii.isWhite(c)) ||
```

Comment:
`is null` is the more idiomatic D way to check for empty strings.

Reply:
Thanks for review, I will update the patch to use null and refresh the patch.

Reply:
`is null` is most definitely _not_ the way to check for empty strings. That's how you check that a string is actually `null`. A `null` string is empty, but an empty string isn't necessarily null. The idiomatic way to check for empty is the `empty` function.

Reply:
Yes, please don't use null to check for empty string, as it will introduce subtle bugs where empty strings with a non-null `.ptr` will not get recognized as empty.  Either use `.empty` or do an explicit length check `chars.length == 0`.